File: data/multilang_comparison/program_julia/src/func39.jl
include("program_julia_head.jl")
function func39(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array56 = nothing
   if pCounter > 1
       array56 = vars.data[pCounter]
       pCounter -= 1
       array56.refC += 1
   else
       array56 = JArray(zeros(Int, 143), 143, 1, 56)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array56)
   params0.size += 1
   array57 = func41(params0, loopsFactor);
   array56.refC -= 1
   if array56.refC == 0
       empty!(array56.data)
       array56 = nothing
   end
   return array57;
end

File: data/multilang_comparison/program_cpp_array/src/func39.cpp
#include "program_cpp_array.hpp" 
Array* func39(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array56;
   if (pCounter > 0) {
      array56 = vars->data[--pCounter];
      array56->refC++;
      DEBUG_COPY(array56->id);
   } else {
      array56 = new Array();
      array56->size = 143;
      array56->refC = 1;
      array56->id = 56;
      array56->data = new unsigned int[array56->size]();
      DEBUG_NEW(array56->id);
   }
   Array_param params0;
   params0.size = 1;
   Array* data_params0[1];
   params0.data = data_params0;
   params0.data[0] = array56;
   Array* array57 = func41(&params0, loopsFactor);
   DEBUG_RETURN(array57->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   array56->refC--;
   if(array56->refC == 0) {
      delete[] array56->data;
      array56->data = nullptr;
      DEBUG_FREE(array56->id);
      delete array56;
   }
   return array57;
}

